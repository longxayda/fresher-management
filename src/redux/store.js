import {configureStore} from "@reduxjs/toolkit";
import attendTableOneTraineeSlice from "./attendanceManagementSlice/attendTableOneTraineeSlice";

// import ticketClassSlice from "./attendanceManagementSlice/ticketClassSlice";
import diligentSummarySlice from "./attendanceManagementSlice/diligentSummarySlice";
import leaveRequestSlice from "./attendanceManagementSlice/leaveRequestSlice";
import checkboxFRReducer from "./fresherManageSlice/checkboxFresherSlice";
import evaluationSlice from "./auditorsSlice/evaluationSlice";
import resourceSlice from "./auditorsSlice/resourceSlice";
import auditorSlice from "./auditorsSlice/auditorSlice";
import manageFRReducer from "./fresherManageSlice/fresherManagementSlice";
import bonusPenaltyReducer from "./fresherManageSlice/penaltySlice";

import classSlice from "./classManagementSlice/classSlice";
import classDetailsSlice from "./classManagementSlice/classDetailsSlice";
import learningPathSlice from "./classManagementSlice/learningPathSlice";
import syllabusSlice from "./classManagementSlice/syllabusSlice";
import moduleSlice from "./classManagementSlice/moduleSlice";
import topicSlice from "./classManagementSlice/topicSlice";

import scoreSlice from "./fresherManageSlice/scoreSlice";
import topicMarkSlice from "./fresherManageSlice/topicMarkSlice";
import scheduleSlice from "./auditorsSlice/scheduleSlice";
import createNewAuditSlice from "./auditorsSlice/createNewAuditSlice";
import adminManageAttendanceReducer from "./attendanceManagementSlice/adminManageAttendanceSlice";
import adminTicketReducer from "./attendanceManagementSlice/adminTicketSlice";

import traineeAttendanceTableSlice from "./attendanceManagementSlice/traineeAttendanceTableSlice";
import authSlice from "./authSlice/authSlice";
import roleSlice from "./roleSlice";
import UserListSlice from "./UserListSlices/UserListSlice";
import attendanceSlice from "./attendanceManagementSlice/attendanceSlice";
import traineeUserProfileSlice from "./userProfileSlice/traineeUserProfileSlice";
import adminUserProfileSlice from "./userProfileSlice/adminUserProfileSlice";

import AddNewReportTableSlice from "components/WeeklyReportTables/addNewReportTableSlice";
import DetailReportTableSlice from "components/WeeklyReportTables/detailReportTableSlice";
import ReportListTableSlice from "components/WeeklyReportTables/reportListTableSlice";
import searchSlice from "components/Search/searchSlice";
import ReportModalSlice from "components/ReportModal/reportModalSlice";

import HistoryReportListTableSlice from "components/MonthlyReportTables/historyReportListTableSlice";
import coursesReportSlice from "../pages/MonthlyReportPage/MonthlyHistoryReport/courses/coursesReportSlice";
import financeReportSlice from "../pages/MonthlyReportPage/MonthlyHistoryReport/finance/financeReportSlice";
import detailCoursesReportSlice from "../pages/MonthlyReportPage/MonthlyHistoryReport/courses/detailCourseReportSlice";
import detailFinanceReportSlice from "../pages/MonthlyReportPage/MonthlyHistoryReport/finance/detailFinanceReportSlice";

import DetailDeliveryTableSlice from "components/MonthlyReportTables/detailDeliveryTableSlice";
import DeliveryReportListTableSlice from "components/MonthlyReportTables/deliveryReportListTableSlice";
import NewDeliveryTableSlice from "components/MonthlyReportTables/newDeliveryTableSlice";

import OutputsReportTableSlice from "components/OutputsReportTables/outputsReportTableSlice";
import fresherReportSlice from "../pages/OutputsReportPage/fresher/fresherReportSlice";
import svttReportSlice from "../pages/OutputsReportPage/svtt/svttReportSlice";

import detailOutputsReportSlice from "../pages/OutputsReportPage/detailOutputsReportSlice";
import detailFresherReportSlice from "../pages/OutputsReportPage/fresher/detailFresherReportSlice";
import detailSVTTReportSlice from "../pages/OutputsReportPage/svtt/detailSVTTReportSlice";

import thunk from "redux-thunk";
const store = configureStore({
	reducer: {
		Filters: searchSlice,
		evaluation: evaluationSlice,
		resource: resourceSlice,
		manageFR: manageFRReducer,
		checkboxFR: checkboxFRReducer,
		attendTableOneTrainee: attendTableOneTraineeSlice,
		// traineeTabTable: TraineeTabTableSlice,
		class: classSlice,
		learningPath: learningPathSlice,
		syllabus: syllabusSlice,
		classDetails: classDetailsSlice,
		module: moduleSlice,
		topicModule: topicSlice,
		auth: authSlice,
		role: roleSlice,
		// UserList
		UserList: UserListSlice,
		schedule: scheduleSlice,
		newAudit: createNewAuditSlice,
		auditor: auditorSlice,

		// attendTable: attendTableSlice,
		// ticketClass: ticketClassSlice,
		diligentSummary: diligentSummarySlice,
		leaveRequest: leaveRequestSlice,
		adminManageAttendance: adminManageAttendanceReducer,
		adminTicket: adminTicketReducer,
		attendance: attendanceSlice,
		traineeAttendanceTable: traineeAttendanceTableSlice,
		bonusPenalty: bonusPenaltyReducer,
		score: scoreSlice,
		topic: topicMarkSlice,
		traineeUserProfile: traineeUserProfileSlice,
		adminUserProfile: adminUserProfileSlice,

		ReportList: ReportListTableSlice,
		AddNewReport: AddNewReportTableSlice,
		DetailReport: DetailReportTableSlice,
		Filters: searchSlice,
		ReportModal: ReportModalSlice,

		HistoryReportList: HistoryReportListTableSlice,
		CoursesReport: coursesReportSlice,
		FinanceReport: financeReportSlice,
		DetailCoursesReport: detailCoursesReportSlice,
		DetailFinanceReport: detailFinanceReportSlice,
		DetailDeliveryReport: DetailDeliveryTableSlice,
		DeliveryReportList: DeliveryReportListTableSlice,
		NewDeliveryReport: NewDeliveryTableSlice,

		OutputsReport: OutputsReportTableSlice,
		FresherReport: fresherReportSlice,
		SVTTReport: svttReportSlice,
		DetailFresherReport: detailFresherReportSlice,
		DetailSVTTReport: detailSVTTReportSlice,
		DetailOutputsReport: detailOutputsReportSlice,
	},
	middleware: [thunk],
});

export default store;
