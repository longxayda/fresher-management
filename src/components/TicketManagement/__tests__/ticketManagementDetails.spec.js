import React from 'react';
import Enzyme from 'enzyme'
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import { mount, shallow } from 'enzyme';
import store from 'redux/store';
import { Provider } from "react-redux";
import TicketManagemetTable from 'components/TicketManagement/index'
Enzyme.configure({ adapter: new Adapter() });

describe('Test Ticket Managemet Table', () => {
    let component;

    test('render ticket management table\'s component with EMPTY props', () => {
        const data = [];
        component = shallow(<TicketManagemetTable data = { data } />);
        expect(component.find('.wrap-table-content').exists()).toEqual(true);

    });

    test('render ticket management table\'s component with NULL props', () => {
        const data = null;
        component = shallow(<TicketManagemetTable data = { data } />);
        expect(component.find('.wrap-table-content').exists()).toEqual(false);
    });

    test('render ticket management table\'s component with VALID props', () => {
        const data = [
            {
            id: "1",
            idtrainee: "123456",
            username: "HieuVo",    
            content: "Request for absence from 01/07/2022 to 01/07/2022",
            status: "Pending",
            handled: "Handle",
            }];
        component = shallow(<TicketManagemetTable data = { data } />);
        expect(component.find('.wrap-table-content').exists()).toEqual(true);
    });

    test('render ticket management table\'s component with INVALID ALL props', () => {
        const data = 'temp';
        component = shallow(<TicketManagemetTable data = {data } />);
        expect(component.find('.wrap-table-content').exists()).toEqual(false);
    });
})

describe('Test Ticket Management Details Page', () => {
    let component;
    let data = [
        {
            id: "1",
            idtrainee: "123456",
            username: "HieuVo",    
            content: "Request for absence from 01/07/2022 to 01/07/2022",
            status: "Pending",
            handled: "Handle",
        },
        {
            id: "2",
            idtrainee: "123457",
            username: "NgocNguyen",
            content: "Request for absence from 05/07/2022 to 06/07/2022",
            status: "Approved",
            handled: "Done",
        },
        {
            id: "3",
            idtrainee: "123458",
            username: "ThaoDoan",
            content: "Request for absence from 05/07/2022 to 05/07/2022",
            status: "Pending",
            handled: "Handle",
        },
        {
            id: "4",
            idtrainee: "123457",
            username: "ThaoDoan",
            content: "Request for absence from 05/07/2022 to 05/07/2022",
            status: "Rejected",
            handled: "Done",
        },
        {
            id: "5",
            idtrainee: "123459",
            username: "NhuY",
            content: "Request for absence from 05/07/2022 to 05/07/2022",
            status: "Approved",
            handled: "Done",
        },
        {
            id: "6",
            idtrainee: "123456",
            username: "Hieuuu",
            content: "Request for absence from 22/07/2022 to 22/07/2022",
            status: "Pending",
            handled: "Handle",
        },
    ];
    beforeEach(() => {
        component = mount(<Provider store = { store }> <TicketManagemetTable data = {data}/> </Provider>);
    });

    test('has the correct class', () => {
        expect(component.find('.wrap-table-content').exists()).toEqual(true);
    });

    test('render Ticket Management Details Page\'s component correctly', () => {
        component = mount(<Provider store = { store }> <TicketManagemetTable data = {data} /> </Provider>);
        expect(component.html()).toMatchSnapshot();
    });
})