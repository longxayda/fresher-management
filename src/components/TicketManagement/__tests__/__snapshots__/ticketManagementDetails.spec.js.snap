// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Ticket Management Details Page render Ticket Management Details Page's component correctly 1`] = `" <div class=\\"wrap-table-content\\"><div class=\\"container\\"><div class=\\"d-flex align-items-center justify-content-between pb-3\\"><div class=\\"d-flex align-items-center justify-content-start\\"><div class=\\"attendance-select-boxes\\"><select class=\\"me-1\\" style=\\"width: 15rem; background-color: rgb(255, 255, 255); border: 1px solid #e3e3e3; border-radius: 4px; color: rgb(86, 86, 86); padding: 8px 12px; height: 40px; box-shadow: none;\\"><option value=\\"Choose class\\">Choose class</option></select><select style=\\"width: 10rem; background-color: rgb(255, 255, 255); border: 1px solid #e3e3e3; border-radius: 4px; color: rgb(86, 86, 86); padding: 8px 12px; height: 40px; box-shadow: none;\\"><option value=\\"Choose class\\">Choose month</option><option value=\\"null-null\\">null-null</option></select></div></div><div class=\\"d-flex align-items-center justify-content-end\\"><div class=\\"me-3\\"><label class=\\"mb-0\\">Search: </label></div><div class=\\"me-3\\"><input type=\\"search\\" class=\\"form-control\\" value=\\"\\"></div><div><button type=\\"button\\" disabled=\\"\\" style=\\"color: white;\\" class=\\"btn-fill  btn btn-primary\\">Accept all <svg aria-hidden=\\"true\\" focusable=\\"false\\" data-prefix=\\"fas\\" data-icon=\\"check\\" class=\\"svg-inline--fa fa-check \\" role=\\"img\\" xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 448 512\\"><path fill=\\"currentColor\\" d=\\"M438.6 105.4C451.1 117.9 451.1 138.1 438.6 150.6L182.6 406.6C170.1 419.1 149.9 419.1 137.4 406.6L9.372 278.6C-3.124 266.1-3.124 245.9 9.372 233.4C21.87 220.9 42.13 220.9 54.63 233.4L159.1 338.7L393.4 105.4C405.9 92.88 426.1 92.88 438.6 105.4H438.6z\\"></path></svg></button><button type=\\"button\\" disabled=\\"\\" style=\\"color: white;\\" class=\\"btn-fill ms-1 btn btn-danger\\">Reject all <svg aria-hidden=\\"true\\" focusable=\\"false\\" data-prefix=\\"fas\\" data-icon=\\"xmark\\" class=\\"svg-inline--fa fa-xmark \\" role=\\"img\\" xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 320 512\\"><path fill=\\"currentColor\\" d=\\"M310.6 361.4c12.5 12.5 12.5 32.75 0 45.25C304.4 412.9 296.2 416 288 416s-16.38-3.125-22.62-9.375L160 301.3L54.63 406.6C48.38 412.9 40.19 416 32 416S15.63 412.9 9.375 406.6c-12.5-12.5-12.5-32.75 0-45.25l105.4-105.4L9.375 150.6c-12.5-12.5-12.5-32.75 0-45.25s32.75-12.5 45.25 0L160 210.8l105.4-105.4c12.5-12.5 32.75-12.5 45.25 0s12.5 32.75 0 45.25l-105.4 105.4L310.6 361.4z\\"></path></svg></button></div></div></div><div style=\\"overflow: auto;\\" class=\\"pt-2 row\\"><div class=\\"container\\"><pre class=\\"chorma\\"><table role=\\"table\\" class=\\"table table-striped table-hover\\"><thead><tr role=\\"row\\"><th colspan=\\"1\\" role=\\"columnheader\\"><div class=\\"heading\\" style=\\"font-size: 16px; font-weight: 600; justify-content: center; text-align: center; width: 100%;\\">&nbsp;</div></th><th colspan=\\"1\\" role=\\"columnheader\\"><div class=\\"heading\\" style=\\"font-size: 16px; font-weight: 600; justify-content: center; text-align: center; width: 100%;\\">Data is loading, please wait...</div></th></tr><tr role=\\"row\\"><th colspan=\\"1\\" role=\\"columnheader\\"><div class=\\"heading\\" style=\\"font-size: 16px; font-weight: 600; justify-content: center; text-align: center; width: 100%;\\"><div><input type=\\"checkbox\\" title=\\"Toggle All Current Page Rows Selected\\" style=\\"cursor: pointer;\\"></div></div></th><th colspan=\\"1\\" role=\\"columnheader\\"><div class=\\"heading\\" style=\\"font-size: 16px; font-weight: 600; justify-content: center; text-align: center; width: 100%;\\"></div></th></tr></thead><tbody role=\\"rowgroup\\"></tbody></table></pre></div></div><div class=\\"row\\"><div class=\\"pagination d-flex align-items-center justify-content-center col\\"><button type=\\"button\\" disabled=\\"\\" class=\\"btn-fill me-3 btn btn-primary\\">&lt;&lt;</button><button type=\\"button\\" disabled=\\"\\" class=\\"btn-fill me-3 btn btn-primary\\">&lt;</button><button type=\\"button\\" disabled=\\"\\" class=\\"btn-fill me-3 btn btn-primary\\">&gt;</button><button type=\\"button\\" disabled=\\"\\" class=\\"btn-fill me-5 btn btn-primary\\">&gt;&gt;</button><span class=\\"me-3 ml-5\\">Page<strong>1 of 0</strong></span><span class=\\"me-3\\"> Go to page: </span><input style=\\"width: 100px;\\" type=\\"number\\" class=\\"me-3 form-control\\" value=\\"1\\"><div class=\\"\\"><select class=\\"form-control\\"><option value=\\"10\\">Show 10</option><option value=\\"20\\">Show 20</option><option value=\\"30\\">Show 30</option><option value=\\"40\\">Show 40</option><option value=\\"50\\">Show 50</option></select></div><br></div></div></div></div> "`;
